RationalNumber.cs

```diff
@@ -105,29 +100,34 @@ public override int GetHashCode()
             if (a.Denominator == b.Denominator)
                 return new(a.Numerator + b.Numerator, a.Denominator);
             else
-                return new(a.Numerator * b.Denominator + b.Numerator * b.Denominator, a.Denominator * b.Denominator);
+                return new(a.Numerator * b.Denominator + b.Numerator * a.Denominator, a.Denominator * b.Denominator);
```

Comment:
```suggestion
                return (double)a.Numerator / a.Denominator > (double)b.Numerator / b.Denominator;
```
Можно же всегда получить вещественное число и сравнить?